### ISSUE TITLE
Cryptopia signing failure

### ISSUE THREAD

**Issue by HughMacdonald** [nlsdfnbch/bitex#68]
### Issue with authentication signing for Cryptopia under 1.1.3 (and 2.0)

# Steps to reproduce

With 1.1.3:

1)

    python3 -c "import bitex;c = bitex.Cryptopia(key='<key>', secret='<secret>' );print(c.balance().json())"

2)

    python3 -c "import bitex;c = bitex.Cryptopia(key='<key>', secret='<secret>' );print(c.balance(Currency='BTC')).json()"

With 2.0:

3)

    python3 -c "import bitex;c = bitex.Cryptopia(key='<key>', secret='<secret>' );print(c.wallet().json())"

4)

    python3 -c "import bitex;c = bitex.Cryptopia(key='<key>', secret='<secret>' );print(c.wallet(Currency='BTC').json())"

# Actual results

1) 

    ERROR:bitex.utils:return_api_response(): Error during call to balance((<bitex.interfaces.cryptopia.Cryptopia object at 0x10ef936a0>,), {})
    Traceback (most recent call last):
      File "/usr/local/lib/python3.5/site-packages/bitex/utils.py", line 32, in wrapper
        r = func(*args, **kwargs)
      File "/usr/local/lib/python3.5/site-packages/bitex/interfaces/cryptopia.py", line 81, in balance
        return self.private_query('GetBalance', params=kwargs)
      File "/usr/local/lib/python3.5/site-packages/bitex/interfaces/cryptopia.py", line 30, in private_query
        return self.query('POST', endpoint, authenticate=True, **kwargs)
      File "/usr/local/lib/python3.5/site-packages/bitex/api/REST/api.py", line 107, in query
        method_verb, *args, **kwargs)
      File "/usr/local/lib/python3.5/site-packages/bitex/api/REST/rest.py", line 369, in sign
        md5 = base64.b64encode(hashlib.md5().updated(post_data).digest())
    AttributeError: '_hashlib.HASH' object has no attribute 'updated'
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/usr/local/lib/python3.5/site-packages/bitex/utils.py", line 32, in wrapper
        r = func(*args, **kwargs)
      File "/usr/local/lib/python3.5/site-packages/bitex/interfaces/cryptopia.py", line 81, in balance
        return self.private_query('GetBalance', params=kwargs)
      File "/usr/local/lib/python3.5/site-packages/bitex/interfaces/cryptopia.py", line 30, in private_query
        return self.query('POST', endpoint, authenticate=True, **kwargs)
      File "/usr/local/lib/python3.5/site-packages/bitex/api/REST/api.py", line 107, in query
        method_verb, *args, **kwargs)
      File "/usr/local/lib/python3.5/site-packages/bitex/api/REST/rest.py", line 369, in sign
        md5 = base64.b64encode(hashlib.md5().updated(post_data).digest())
    AttributeError: '_hashlib.HASH' object has no attribute 'updated'

2) Same as 1)

3) As expected

4)

    {'Error': 'Signature does not match request parameters.', 'Success': False}

# Expected Results

1) Balance of all currencies in wallets
2) Balance of BTC wallet
3) Balance of all currencies in wallets
4) Balance of BTC wallet

# Reproducibility

This error is reproducible every time

# Additional Information

If I copy the inner workings of sign() from the 2.0 branch, it works. This gives me a sign() method of:

    def sign(self, uri, endpoint, endpoint_path, method_verb, *args, **kwargs):
        nonce = self.nonce()
        try:
            params = kwargs['params']
        except KeyError:
            params = {}

        post_data = json.dumps(params)

        # generate signature
        md5 = hashlib.md5()
        md5.update(post_data.encode('utf-8'))
        request_content_b64_string = base64.b64encode(md5.digest()).decode('utf-8')
        signature = (self.key + 'POST' +
                     urllib.parse.quote_plus(uri).lower() +
                     nonce + request_content_b64_string)

        hmac_sig = base64.b64encode(hmac.new(base64.b64decode(self.secret),
                                             signature.encode('utf-8'),
                                             hashlib.sha256).digest())
        header_data = 'amx ' + self.key + ':' + hmac_sig.decode('utf-8') + ':' + nonce

        # Update req_kwargs keys
        headers = {'Authorization': header_data,
                                 'Content-Type': 'application/json; charset=utf-8'}

        return uri, {'headers': headers, 'data': post_data}

This then works for both 1. and 2. in 1.1.3

I can submit a PR if you'd like. It's my first time interacting with this repository so I wanted to put it in as an issue first.

**Comment by nlsdfnbch** [comment]
Hey Hugh!
Thanks for submitting this! 
Please do submit a PR - no need in reinventing the wheel if you've already done so :) The help is much appreciated.

As for passing a currency in `2.0`, I'll look into that. Do the other private endpoints work for you on that branch?

**Comment by HughMacdonald** [comment]
You're welcome

Sure - will do that now.

I didn't try any further in 2.0, as I got it working in 1.1.3. I was starting with the simplest thing I could do through the private API.

I've got a few things I'm trying to do here - I have 5 exchanges I want to be able to use, only 3 of which you support at the moment, plus one of the exchanges that you do support (Kraken) has a couple of NotImplementedErrors in there (`withdraw` and `deposit_address`). I want to see if I can implement the missing endpoints in Kraken, and then will see if I can add the additional 2 exchanges, if you'd be open to that?
